mutation/mutation_fragment_v2.hh

```diff
@@ -370,6 +370,31 @@ private:
 
 std::ostream& operator<<(std::ostream&, mutation_fragment_v2::kind);
 
+template <> struct fmt::formatter<mutation_fragment_v2::kind> : fmt::formatter<std::string_view> {
+    template <typename FormatContext>
+    auto format(const mutation_fragment_v2::kind& k, FormatContext& ctx) const {
```

Comment:
since `mutation_fragment_v2::kind` is an enum, whose underlying type is `int`, so we can just pass it by value without using reference. more performance this way.

also, if we don't need to support the use cases like `fmt::to_string()`, we could just define the implementation in .cc file and use the more ugly way to declare the `format()` member function: like
```c++
auto format(mutation_fragment_v2::kind, fmt::format_context& ctx) const -> decltype(ctx.out());
```
and define its implementation in the .cc file. the downside is that we only support the `fmt::format_context`. but it would suffice in the use case of formatting it to seastar's logger.

Reply:
I'll fix the first thing, I don't understand the second thing. I don't know what is fmt::format_context. I copied the templated example from fmt's official documentation :-(